lib/prompts/createApiSamplePrompt.ts

```diff
@@ -1,52 +1,85 @@
 
 const i18nKey = 'lib.prompts.createApiSamplePrompt';
 
-const getSampleTypesPrompt = choices => ({
-  type: 'rawlist',
-  name: 'sampleType',
-  message: i18n(`${i18nKey}.selectApiSampleApp`),
-  choices: choices.map(choice => ({
-    name: `${choice.name} - ${choice.description}`,
-    value: choice.id,
-  })),
-  validate: input => {
-    return new Promise(function(resolve, reject) {
-      if (input.length > 0) {
-        resolve(true);
-      }
-      reject(i18n(`${i18nKey}.errors.apiSampleAppRequired`));
-    });
-  },
-});
+type SampleChoice = {
+  name: string;
+  description: string;
+  id: string;
+  languages: string[];
+};
+
+type SampleConfig = {
+  samples: SampleChoice[];
+};
+
+type createApiSamplePromptResponse = {
+  sampleType?: string;
+  sampleLanguage?: string;
+};
 
-const getLanguagesPrompt = choices => ({
-  type: 'rawlist',
-  name: 'sampleLanguage',
-  message: i18n(`${i18nKey}.selectLanguage`),
-  choices: choices.map(choice => ({
-    name: choice,
-    value: choice,
-  })),
-  validate: input => {
-    return new Promise(function(resolve, reject) {
-      if (input.length > 0) {
-        resolve(true);
-      }
-      reject(i18n(`${i18nKey}.errors.languageRequired`));
-    });
-  },
-});
+function getSampleTypesPrompt(
+  choices: SampleChoice[]
+): PromptConfig<createApiSamplePromptResponse> {
+  return {
+    type: 'rawlist',
+    name: 'sampleType',
+    message: i18n(`${i18nKey}.selectApiSampleApp`),
+    choices: choices.map(choice => ({
+      name: `${choice.name} - ${choice.description}`,
+      value: choice.id,
+    })),
+    validate: function(input?: string): Promise<PromptOperand> {
+      return new Promise<PromptOperand>(function(resolve, reject) {
+        if (input && input.length > 0) {
+          resolve(true);
+        } else {
+          reject(i18n(`${i18nKey}.errors.apiSampleAppRequired`));
+        }
+      });
+    },
+  };
+}
 
-const createApiSamplePrompt = async samplesConfig => {
+function getLanguagesPrompt(
+  choices: string[]
+): PromptConfig<createApiSamplePromptResponse> {
+  return {
+    type: 'rawlist',
+    name: 'sampleLanguage',
+    message: i18n(`${i18nKey}.selectLanguage`),
+    choices: choices.map(choice => ({
+      name: choice,
+      value: choice,
+    })),
+    validate: function(input: string | undefined): Promise<PromptOperand> {
+      return new Promise<PromptOperand>(function(resolve, reject) {
+        if (input && input.length > 0) {
+          resolve(true);
+        }
+        reject(i18n(`${i18nKey}.errors.languageRequired`));
+      });
+    },
+  };
+}
+
+export async function createApiSamplePrompt(
+  samplesConfig: SampleConfig
+): Promise<createApiSamplePromptResponse> {
   try {
     const { samples } = samplesConfig;
     const sampleTypeAnswer = await promptUser(getSampleTypesPrompt(samples));
     const chosenSample = samples.find(
       sample => sample.id === sampleTypeAnswer.sampleType
     );
+    if (!chosenSample) {
```

Comment:
I think it's safe to do `const { languages } = chosenSample!;` here. We know `chosenSample` will always exist so I don't think the error message will ever be triggered